### PR TITLE
DDF-4176 Update ConfluenceFilterDelegate to handle propertyIsGreaterThan

### PR DESCRIPTION
**Pull request by clockard**
#### What does this PR do?
Update ConfluenceFilterDelegate to handle propertyIsGreaterThan
- Also fix and issue with the TagsFilterDelegate 'not' implementation

#### Who is reviewing it? 

@peterhuffer @emmberk @kcover 
#### Select relevant component teams: 
https://github.com/orgs/codice/teams
#### Ask 2 committers to review/merge the PR and tag them here.

@coyotesqrl
@rzwiefel
@tbatie

#### How should this be tested?

See me for test steps
#### What are the relevant tickets?
[DDF-4176](https://codice.atlassian.net/browse/DDF-4176)
#### Checklist:
- [X] Update / Add Unit Tests

### CODE REVIEW

**Comment by clockard** [comment]
build now

File: catalog/core/catalog-core-commons/src/main/java/ddf/catalog/filter/delegate/TagsFilterDelegate.java [hunk 1]
```diff
@@ -62,7 +62,7 @@ public Boolean or(List<Boolean> operands) {
 
   @Override
   public Boolean not(Boolean operand) {
-    return tags == null ? operand : !operand;
```
Comment (rzwiefel) [inline catalog/core/catalog-core-commons/src/main/java/ddf/catalog/filter/delegate/TagsFilterDelegate.java]: Are we sure about changing this? If I remember it was to fix an issue with not respecting `not` correctly.

**Comment by clockard** [comment]
build now

File: catalog/core/catalog-core-commons/src/main/java/ddf/catalog/filter/delegate/TagsFilterDelegate.java (hunk 1 above)
Comment (clockard) [reply catalog/core/catalog-core-commons/src/main/java/ddf/catalog/filter/delegate/TagsFilterDelegate.java]: Yes. The TagsFilterDelegate is just trying to determine if each clause has the needed metacard-tag property. In the case of the not clause we just want to pass along the result of whatever was in it.

**Comment by clockard** [comment]
build now

**Comment by clockard** [comment]
Canceled the previous build because it was hung up in itests. 
build now

**Comment by clockard** [comment]
build now

**Comment by peterhuffer** [comment]
Successful hero. Verified itests of downstream projects pass locally. Verified Registry publishing, updating and deleting. Verified fanout queries for confluence return results. ✅

**Pull request merged**